### PR TITLE
vc: improvements for testing

### PR DESCRIPTION
**Pull request by vvolkl**
Fixing a couple of issues with this recipe:
- the `build_type` variant is not needed (provided by the default  CMakePackage)
- `vc` should not depend on `virtest@master` which is a moving target that will likely break things. The 1.4 series of vc sets up a specific commit of virtest as a submodule.
  - the virtest dependency *should* be `type=('test')` - virtest just needs to set up `dependent_build_environment` for this to work.
  - However, it seems like virtest is not used outside of vc, has no published releases and is header only, so I don't see the benefit of a dedicated package for it and am removing it
- The patch function as is breaks the 1.3 versions of vc.
  - The main reason those tests are failing is that they require the test data, also set up via submodules. Fetching this data fixed the test suite for me on the configuration mentioned below.
 

I only checked  that the test suite is completely passing for the latest version for linux-ubuntu20.04-x86_64 / gcc@9.3.0. It's also not very lightweight, fetching 300 MB of test data and adding 30min of compilation time, but I guess if you want to test this package it's best to test it as the maintainers intended.

### CODE REVIEW

File: var/spack/repos/builtin/packages/vc/package.py [hunk 1]
```diff
@@ -10,30 +10,38 @@ class Vc(CMakePackage):
...
-
     def cmake_args(self):
-        if self.run_tests:
-            return ['-DBUILD_TESTING=ON']
-        else:
-            return ['-DBUILD_TESTING=OFF']
+        return [self.define('BUILD_TESTING', self.run_tests)]
+
+    @run_before('cmake')
+    def fetch_additional_sources(self):
```
Comment (iarspider) [inline var/spack/repos/builtin/packages/vc/package.py:24]: Can you make these two repos `resource`s?

**Comment by vvolkl** [comment]
Request for merge?  @adamjstewart @alalazo ? Thank you :smiley:

**Review by iarspider — DISMISSED** [review]

File: var/spack/repos/builtin/packages/vc/package.py (hunk 1 above)
Comment (vvolkl) [reply var/spack/repos/builtin/packages/vc/package.py:24]: I've tried to figure out if `resources` can be of type "test", but this seems not (yet) possible. @alalazo @adamjstewart  correct me if I'm wrong!

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/vc/package.py:24]: Not possible as far as I know, but I would still make them resources and just always download them.

Comment (vvolkl) [reply var/spack/repos/builtin/packages/vc/package.py:24]: @adamjstewart Are you sure? The test data has 300MB and is really not needed for most users of spack

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/vc/package.py:24]: Hmm, that is quite large. I assume it isn't possible to add it as a separate package with a test dependency? @scheibelp @tldahlgren how hard would it be to add a flag to `resource` that only downloads the resource when testing?

File: var/spack/repos/builtin/packages/vc/package.py [hunk 2]
```diff
@@ -10,30 +10,38 @@ class Vc(CMakePackage):
...
-        tests.filter(';AVX2', '')
-        tests.filter('.*logarithm.*', '')
-        tests.filter('.*trigonometric.*', '')
-
     def cmake_args(self):
-        if self.run_tests:
-            return ['-DBUILD_TESTING=ON']
-        else:
-            return ['-DBUILD_TESTING=OFF']
+        return [self.define('BUILD_TESTING', self.run_tests)]
```
Comment (cgmb) [inline var/spack/repos/builtin/packages/vc/package.py]: You don't need this. [It's already handled by CMakePackage](https://github.com/spack/spack/blob/18c2f1a57a69224748e091e912851317d8a00eea/lib/spack/spack/build_systems/cmake.py#L179).

**Comment by alalazo** [comment]
@vvolkl What is the state of this PR?

**Comment by vvolkl** [comment]
@alalazo Ready to be merged, I'd say. I haven't checked if `resource` has been adapted for the usecase here, but it's still a big improvement.

**Review by alalazo — APPROVED** [review]

**Pull request merged**